### PR TITLE
WIP: tfm-build

### PR DESCRIPTION
**Pull request by sharvit**
## Updated Packages

* [ ] root
* [ ] @theforeman/babel
* [x] @theforeman/builder
* [ ] @theforeman/env
* [ ] @theforeman/vendor
* [ ] @theforeman/vendor-dev
* [ ] @theforeman/vendor-core

## PR Type

* [ ] Bugfix
* [x] Feature
* [ ] Code style update (whitespace, formatting, missing semicolons, etc.)
* [ ] Refactoring (no functional changes, no api changes)
* [ ] Build related changes
* [ ] CI related changes
* [ ] Documentation content changes
* [ ] Other… Please describe:

## Description

Still work in progress, the goal is to be able to build foreman.

## Does this PR introduce a breaking change?

* [ ] Yes
* [x] No

## Does this PR fixes open issues?

* [ ] Yes
* [x] No

## Do you use this PR in another repository?

* [x] Yes - https://github.com/theforeman/foreman/pull/7214
* [ ] No

### CODE REVIEW

File: packages/builder/config/createWebpackRules.js [hunk 1]
```diff
@@ -0,0 +1,28 @@
+const MiniCssExtractPlugin = require('mini-css-extract-plugin');
+
+const createWebpackRules = mode => [
```
Comment (xprazak2) [inline packages/builder/config/createWebpackRules.js]: `mode` is not (yet) used

**Comment by xprazak2** [comment]
I am not sure how I would fix this...

builder-err

**Comment by xprazak2** [comment]
If 'config' is not supplied as arg:

required-config

it could be required or we might fall back to a default version if it makes sense.

**Comment by sharvit** [comment]
Thanks @xprazak2 :+1:

File: packages/builder/config/plugins/createWebpackCommonPlugins.js [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+  webpackManifestFilename,
+} = require('../../lib/webpackHelpers');
+
+const createWebpackCommonPlugins = (mode, userConfig) => {
+  const supportedLanguagesRE = createSupportedLanguagesRE(userConfig.locales);
+
+  return [
+    new WebpackForemanVendorPlugin({ mode }),
+    new StatsWriterPlugin({
+      filename: webpackManifestFilename(),
```
Comment (xprazak2) [inline packages/builder/config/plugins/createWebpackCommonPlugins.js]: I think you need to pass `mode` to `webpackManifestFilename`

Comment (xprazak2) [inline packages/builder/config/plugins/createWebpackCommonPlugins.js]: I think you need to pass an argument for plugin names to resolve correctly.

**Comment by xprazak2** [comment]
Awesome work, I managed to get a green build today (with a couple of size warnings) for core:

green-builder

**Comment by sharvit** [comment]
Closing it and I will open a new one soon as this one is too old

**Pull request closed without merging**